src/config/datadog.rs

```diff
@@ -28,14 +30,15 @@ impl Default for Options {
         Self {
             enabled: default_enabled(),
             api_key: None,
+            configuration: "".to_owned(),
```

Comment:
I went with `config_id` here in the generated metrics (I believe you originally suggested this).  Could be persuaded to go for `config_uuid` (maybe better).   I prefer those over just `configuration` to get across the idea that we're looking for an id :two_cents 
```suggestion
            config_id: "".to_owned(),
```

Reply:
Another :two_cents, we should try to call it consistently the same thing in the UI and Vector.  Easy to update the generated metrics to whatever the final is 👍

Reply:
Thinking about this some more, I would be tempted to call this field `configuration_key`.

The reason I avoided "id" was that the actual _purpose_ of the field feels like an implementation detail of Pipelines, and not something the user should be concerned with. The user doesn't need to know this indicates a row in a table.

Similar for "_uuid". Along with being an ID, this now restricts us to always using a particular format/data type. If we later change this, _uuid_ could be moot.

"key", OTOH, feels more opaque. It could be an ID, a UUID, or something else entirely -- the user doesn't need to know.

What do you think?

Reply:
A minor detail, but I also like the fact we're already using it for `api_key`, which looks cohesive in a config:

```toml
[datadog]
api_key = "xxx"
configuration_key = "zzz"
```

Reply:
Sounds good to me.  I've updated the wiki and making this `configuration_key` in our metrics today.  🚀